pkg/apis/machine/v1alpha1/machineclass_types.go

```diff
@@ -58,3 +61,13 @@ type MachineClassList struct {
 	metav1.ListMeta `json:"metadata,omitempty"`
 	Items           []MachineClass `json:"items"`
 }
+
+// ExpectedNodeDetails contains subfields the track all scale from zero resources
+type ExpectedNodeDetails struct {
```

Comment:
Also if you like you can change the name to `ExpectedNodeInfo` or something..your call